### PR TITLE
update deprecate dependency

### PR DESCRIPTION
**Pull request by vincentgoat**
Signed-off-by: vincentgoat <linguohui1@huawei.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
3 projects in KubeEdges dependency tree were found to be deprecated or unmaintained.
- github.com/golang/protobuf
- github.com/gorilla/mux (see https://github.com/gorilla/mux/issues/659) 
- github.com/paypal/gatt (see https://github.com/paypal/gatt/issues/75). gatt is only used for testing.

2nd and 3rd issues are solved in this PR.
As for now, github.com/golang/protobuf is remain used by github.com/container-storage-interface/spec/lib/go/csi, this might need a deeper analysis on how to solve. /help

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by gy95** [comment]
Do you mean to remove or replace the two packages? there're still some other codes using the two packages. this patch don't remove these packages totally

**Comment by vincentgoat** [comment]
> Do you mean to remove or replace the two packages? there're still some other codes using the two packages. this patch don't remove these packages totally

Yes, replace totally. Which are the other codes?

**Comment by gy95** [comment]
For example:
https://github.com/kubeedge/kubeedge/blob/f4420cd3a354d8a8f9ef02fd8d195ca2d0eeab3a/edge/test/cloudhub/stub.go#L8
IMHO, if we've already removed dependency package totally, we need run `make verify-vendor` and `make verify-vendor-licenses` to update vendor

File: cloud/pkg/cloudhub/servers/httpserver/server.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/emicklei/go-restful"
 	"github.com/golang-jwt/jwt"
-	"github.com/gorilla/mux"
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/httpserver/server.go:33]: Is there other part using `github.com/gorilla/mux` in KubeEdge?

**Comment by fisherxu** [comment]
> IMHO, if we've already removed dependency package totally, we need run make verify-vendor and make verify-vendor-licenses to update vendor

Maybe other vender will still use it.. @gy95 @vincentgoat

File: cloud/pkg/cloudhub/servers/httpserver/server.go (hunk 1 above)
Comment (gy95) [reply cloud/pkg/cloudhub/servers/httpserver/server.go:33]: only edge/test/cloudhub/stub.go
https://github.com/kubeedge/kubeedge/blob/f4420cd3a354d8a8f9ef02fd8d195ca2d0eeab3a/edge/test/cloudhub/stub.go#L8

And github.com/docker/distribution still use it

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**